libres/lib/job_queue/lsf_driver.cpp

```diff
@@ -959,24 +927,61 @@ void lsf_driver_free_job(void *__job) {
     lsf_job_free(job);
 }
 
+namespace lsf {
+std::vector<std::string> split_string(std::string str, std::string delimiter) {
```

Comment:
Should move such a generic function to a generic location: `ert/util/string.hpp` (check that this doesn't overlap with libecl)

If you wanna be fancy, you could make this take a functor so you wouldn't need to create a `std::vector`. The following is entirely no-copy due to `std::string_view` (UB Danger: If the functor changes the original string then the entire thing is UB. So this should be documented):

```c++
template <typename Func>
void split(std::string_view str, char delimiter, Func&& func) {
    size_t pos = 0, next_pos = 0;

    while ((next_pos = str.find(delimiter, pos)) != std::string::npos) {
        func(str.substr(pos, next_pos - pos));
        pos = next_pos + delimiter.size();
    }
    func(str.substr(pos, str.size() - pos));
}
```

Also we don't need to substring search when searching for a single character.

Reply:
Indeed, I looked into this as there are functions for doing roughly these operations in several places, but they seem to depend on slightly diverging behavior so I didn't have time to finish this before parental leave.